Automattic/wp-calypso
# Manage All Domains: Fix table alignment

Fixes https://github.com/Automattic/dotcom-forge/issues/2969

## Proposed Changes

* Add "Actions" column label to fix alignment.

Before:
<img width="1083" alt="Screen Shot 2023-07-07 at 3 42 49 PM" src="https://github.com/Automattic/wp-calypso/assets/1689238/fc3abf88-4e03-464d-a8a3-2f5075f88717">

After:
<img width="1075" alt="Screen Shot 2023-07-07 at 3 42 34 PM" src="https://github.com/Automattic/wp-calypso/assets/1689238/7fd05e71-4770-4949-be86-2ff383f03411">


## Testing Instructions



* Go to /domains/manage
* Check that the table data looks aligned with the header.
* Change the domains dropdown to All domains / My domains / etc. to check that the table is still aligned.
* Check the mobile view of the table.
* Check additional browsers.
* Go to /domains/manage/{site}
* Check that the table and headers are aligned and spaced out correctly.

## Pre-merge Checklist
